pkg/virt-handler/vm.go

```diff
@@ -1299,6 +1385,12 @@ func (d *VirtualMachineController) processVmCleanup(vmi *v1.VirtualMachineInstan
 
 	vmiId := string(vmi.UID)
 
+	// Don't process if the vmi uid is nil
```

Comment:
@vatsalparekh If we exit here, vmGracefulShutdownTriggerClear won't be called but it doesn't need UID, maybe we should first do the graceful shutdown cleanup and then check for UID validity.

Reply:
Copying from the chat - 
IIUC, we can get to the situation there is no uid on the vmi only if the vmi and the domain don't exist on the virt-handler informers and also not in the GhostRecordCache. The vmi is removed from the GhostRecordCache during the cleanup. So it means cleanup was already done. So I think it is safe to return from the [execute](https://github.com/kubevirt/kubevirt/blob/ac98c6b523cd88223173c89c2e8dbf244d3744f4/pkg/virt-handler/vm.go#L1240).

@enp0s3 IIUC, if we get to here with no uid, it means the cleanup was already invoked.